core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1056,9 +1056,7 @@ final class Stream[+F[_], +O] private (private val free: FreeC[Algebra[Nothing,
     Stream
       .getScope[F2]
       .flatMap { scope =>
-        Stream.eval(F2.start(haltOnSignal.flatMap(scope.interrupt))).flatMap { fiber =>
-          this.onFinalize(fiber.cancel)
-        }
+        Stream.supervise(haltOnSignal.flatMap(scope.interrupt)) >> this
```

Comment:
It seemed like the old code was still finalising the fiber. A bit unclear wrt difference now

Reply:
Yeah, this change is gratuitous and unrelated to the fix. I could back it out if you think it's not needed. The real fix is the `F.never -> F.unit` change.

Reply:
I don't mind it being there as a refactor. Just wanted to know if behaviour was changing or not.
I guess as it is, it's slightly surprising since it (seems to) warrant the introduction of a combinator, `supervise`, which isn't used anywhere else.

No real preference here tbh.

Reply:
On second look, this change isn't gratuitous though it doesn't help with the failing test case. The problem here is that a fiber is started and then it separately registered for cancelation. It could have been fixed by replacing eval + flatMap + onFinalize with bracket.

Reply:
ah, true, if `start` succeeds but then things get interrupted the `cancel` finaliser never gets registered, 👍